### PR TITLE
Additions to the system package

### PR DESCRIPTION
**Pull request by amrhassan**

### CODE REVIEW

File: lib/JumpScale/core/system/net.py [hunk 1]
```diff
@@ -1091,6 +1091,36 @@ def updateHostsFile(self,hostsfile,ip,hostname):
...
+                return
+
+        # If reached here then downloading is inevitable
+        j.system.net.download(url, localpath=destination_file_path, username=http_auth_username,
+                              passwd=http_auth_password)
+
+        # Now check if the downloaded file matches the provided checksum
+        if md5_checksum and not j.system.fs.compute_md5_checksum(destination_file_path) == md5_checksum:
+            raise RuntimeError('The provided MD5 checksum did not match that of a freshly-downloaded file!')
+
```
Comment (grimpy) [inline lib/JumpScale/core/system/net.py]: use j.tools.hash.md5 instead

This is close to what used in jpackage.py for web.export could you make it use this?

File: lib/JumpScale/core/system/net.py [hunk 2]
```diff
@@ -1091,6 +1091,36 @@ def updateHostsFile(self,hostsfile,ip,hostname):
         j.system.fs.writeFile(hostsfile, filecontents)
 
 
+    def download_if_non_existent(self, url, destination_file_path, md5_checksum=None,
```
Comment (grimpy) [inline lib/JumpScale/core/system/net.py]: methodname does not fit the rest of JumpScale naming convention use camelCase

**Comment by amrhassan** [comment]
@grimpy *Nudge*

File: lib/JumpScale/core/system/net.py [hunk 3]
```diff
@@ -1091,6 +1091,36 @@ def updateHostsFile(self,hostsfile,ip,hostname):
...
+
+                # On invalid checksum, delete the local file
+                j.system.fs.remove(destination_file_path)
+
+            else:
+                # It exists but no checksum is provided so any existence of the local file suffices.
+                return
+
+        # If reached here then downloading is inevitable
+        j.system.net.download(url, localpath=destination_file_path, username=http_auth_username,
```
Comment (grimpy) [inline lib/JumpScale/core/system/net.py]: self.download?

**Pull request merged**